### PR TITLE
Implement APIs for retrieving follower count and follower details

### PR DESCRIPTION
**Pull request by SAKET687**
This PR introduces new APIs to enhance user engagement by providing follower-related functionalities:

- **GET /followers/count:** Retrieves the total number of followers for a specific user.
- **GET /followers/details:** Fetches detailed information about a user's followers, including their profiles.

These additions aim to improve the overall user experience and enable better interaction with follower data.

**Changes Made:**
- Created API endpoints for follower count and details.

### CODE REVIEW

File: src/Routes/restaurant.route.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ const {
 	restaurantMenuUploadValidation,
 	restaurantMenuUpdateValidation,
 	restaurantMenuDeleteValidation,
+	restaurantAdminFollowersDisplayValidation,
```
Comment (arka1997) [inline src/Routes/restaurant.route.js:15]: can we make this name chnage to "restaurantFollowersDisplayValidation", if this name points to the Restaurant Followers list

File: src/Routes/restaurant.route.js [hunk 2]
```diff
@@ -494,4 +496,59 @@ module.exports = [
 			description: "Display Restaurant's Reviews",
 		},
 	},
+
+	// display all followers of the restaurant
+	{
+		method: "GET",
+		path: "/restaurant-admin-followers",
```
Comment (arka1997) [inline src/Routes/restaurant.route.js:503]: name can be like: restaurant-follower-details. Actually I don't to wantkeep the word "Admin" for Restaurants. I twill create confusion with the Super Admin things.

File: src/Routes/restaurant.route.js (hunk 1 above)
Comment (SAKET687) [reply src/Routes/restaurant.route.js:15]: I used `admin` to distinguish between two follower APIs: one for restaurant admins, which shows user details like user ID, name & email and another for regular users, which only shows the follower's user ID & name.

File: src/Routes/restaurant.route.js (hunk 2 above)
Comment (SAKET687) [reply src/Routes/restaurant.route.js:503]: Ok, will do that.

**Review by arka1997 — APPROVED** [review]

**Review by arka1997 — APPROVED** [review]

**Review by SaiAshish1729 — APPROVED** [review]

**Pull request merged**